### ISSUE TITLE
🚀 Feature: EntityCatalogGraphCard is unusable when having lot of children

### ISSUE THREAD

**Issue by jenil-harness** [backstage/backstage#18444]
### 🔖 Feature description

When a parent node has a lot of immediate children, it takes a lot of time to load and is also unusable.

### 🎤 Context

EntityCatalogGraphCard is unusable when having a lot of immediate children. Adding a way to control the max depth will make it more usable.

### ✌️ Possible Implementation

Adding maxDepth prop for EntityCatalogGraphCard

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by sennyeya** [comment]
The `EntityCatalogGraphCard` already has support for `maxDepth` [here](https://github.com/backstage/backstage/blob/master/plugins/catalog-graph/src/components/CatalogGraphCard/CatalogGraphCard.tsx#L68).

Is this for vertical or horizontal depth? Possibly related https://github.com/backstage/backstage/issues/18061.

**Comment by jenil-harness** [comment]
@sennyeya It is for the horizontal depth (breadth of tree ). 
Yes the issue is the same as https://github.com/backstage/backstage/issues/18061

**Comment by Rugvip** [comment]
Yep gonna close as a duplicate of #18061, let's keep the discussion focused there! 👍
